tests/models/swiftformer/test_modeling_tf_swiftformer.py

```diff
@@ -0,0 +1,276 @@
+            embed_dims=self.embed_dims,
+            mlp_ratio=4,
+            downsamples=[True, True, True, True],
+            hidden_act="gelu",
+            num_labels=self.num_labels,
+            down_patch_size=3,
+            down_stride=2,
+            down_pad=1,
+            drop_rate=0.0,
+            drop_path_rate=0.0,
+            use_layer_scale=True,
+            layer_scale_init_value=1e-5,
+        )
+
+    def create_and_check_model(self, config, pixel_values, labels):
+        model = TFSwiftFormerModel(config=config)
+        result = model(pixel_values)
+        # FIXME: channels_first or last?
+        self.parent.assertEqual(result.last_hidden_state.shape, (self.batch_size, self.embed_dims[-1], 7, 7))
+
+    def create_and_check_for_image_classification(self, config, pixel_values, labels):
+        config.num_labels = self.num_labels
+        model = TFSwiftFormerForImageClassification(config)
+        result = model(pixel_values, labels=labels)
+        self.parent.assertEqual(result.logits.shape, (self.batch_size, self.num_labels))
+
+        model = TFSwiftFormerForImageClassification(config)
+
+        pixel_values = floats_tensor([self.batch_size, self.num_channels, self.image_size, self.image_size])
+        result = model(pixel_values)
+        self.parent.assertEqual(result.logits.shape, (self.batch_size, self.num_labels))
+
+    def prepare_config_and_inputs_for_common(self):
+        (config, pixel_values, labels) = self.prepare_config_and_inputs()
+        inputs_dict = {"pixel_values": pixel_values}
+        return config, inputs_dict
+
+
+@require_tf
+class TFSwiftFormerModelTest(TFModelTesterMixin, PipelineTesterMixin, unittest.TestCase):
+    """
+    Here we also overwrite some of the tests of test_modeling_common.py, as SwiftFormer does not use input_ids, inputs_embeds,
+    attention_mask and seq_length.
+    """
+
+    all_model_classes = (TFSwiftFormerModel, TFSwiftFormerForImageClassification) if is_tf_available() else ()
+
+    pipeline_model_mapping = (
+        {"feature-extraction": TFSwiftFormerModel, "image-classification": TFSwiftFormerForImageClassification}
+        if is_tf_available()
+        else {}
+    )
+
+    fx_compatible = False
+    test_pruning = False
+    test_resize_embeddings = False
+    test_head_masking = False
+    has_attentions = False
+    test_onnx = False
+
+    def setUp(self):
+        self.model_tester = TFSwiftFormerModelTester(self)
+        self.config_tester = ConfigTester(
+            self,
+            config_class=SwiftFormerConfig,
+            has_text_modality=False,
+            hidden_size=37,
+            num_attention_heads=12,
+            num_hidden_layers=12,
+        )
+
+    def test_config(self):
+        self.config_tester.run_common_tests()
+
+    @unittest.skip(reason="TFSwiftFormer does not use inputs_embeds")
+    def test_inputs_embeds(self):
+        pass
+
+    def test_model_common_attributes(self):
```

Comment:
All of the methods here which are identical to other implementations should have `# Copied from` headers

Reply:
I copied the method from the original file and did the necessary modifications to run it on TensorFlow, is it still necessary to add a `# Copied from` comment? I thought it would be expected that the TensorFlow version is an adaptation of the PyTorch code.

Reply:
`# Copied from` comments would be for copying from equivalent TF tests, rather than pytorch ones. In this case,  it looks equivalent to the test in `test_modeling_tf_deit.py`

Reply:
This wasn't addressed

Reply:
The `test_modeling_tf_deit.py` has an additional `self.assertIsInstance(model.get_input_embeddings(), (keras.layers.Layer))` that is not present in this version.

I tried adding that line but got an error `AttributeError: 'TFSwiftFormerMainLayer' object has no attribute 'get_input_embeddings'`. Do you have any suggestions? Should I add the method or keep the test without the `# Copied from` comment?

Reply:
Hi @amyeroberts , just pinging!

Reply:
If there's a difference between the tests, and adding the `get_input_embeddings` method doesn't make sense (as in this case) then it's OK to skip for this one

Reply:
Addressed here: https://github.com/huggingface/transformers/pull/23342/commits/3a2b6fd54834ecdc98d6147c1d40f576a04e24b2